src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs

```diff
@@ -16,56 +16,142 @@ namespace Analyzer.Utilities
 {
     internal static partial class AnalyzerOptionsExtensions
     {
-        private static readonly ConditionalWeakTable<AnalyzerOptions, CategorizedAnalyzerConfigOptions> s_cachedOptions
-            = new ConditionalWeakTable<AnalyzerOptions, CategorizedAnalyzerConfigOptions>();
+        private static readonly ConditionalWeakTable<AnalyzerOptions, ICategorizedAnalyzerConfigOptions> s_cachedOptions
+            = new ConditionalWeakTable<AnalyzerOptions, ICategorizedAnalyzerConfigOptions>();
         private static readonly ImmutableHashSet<OutputKind> s_defaultOutputKinds =
             ImmutableHashSet.CreateRange(Enum.GetValues(typeof(OutputKind)).Cast<OutputKind>());
 
+        private static TValue GetOptionWithConflictHandling<TValue>(ISymbol symbol, Func<SyntaxTree, TValue> getOptionForTree, TValue conflictValue)
+        {
+            var value = getOptionForTree(symbol.Locations[0].SourceTree);
+
+            // Handle conflicts from option value in different partial declaration of the symbol.
```

Comment:
I have been in two minds of whether we should be doing conflict resolution for differing option values across partial declaration of a symbol. After putting some thought, I am going to remove the conflict resolution logic with following reasoning:

1. This is a highly unlikely scenario, especially given most of the non-generated type definitions will be in the same folder and likely have same options.
2. Conflict value chosen might still not be intuitive for the end user, as it will end up being the default value for most cases. The only real solution here is to report a user facing diagnostic about conflicting option values, but that needs much more wholesale repo wide changes, which seems too much work for a corner case scenario.
3. Performance: We will unnecessarily compute option values for multiple trees, which will end up being identical. Additionally, we will end up allocating a delegate for each invocation of this helper.

I will file a feature request for handling conflict resolution, in case we decide to support it in future.